routes/api.php

```diff
@@ -16,22 +16,30 @@
 Route::get('/', 'StartupController@index');
 
 // Auth system
-Route::post('/login', 'LoginController@login')->name('login');
-Route::post('/register', 'AuthController@store')->name('register');
-Route::post('/logout', 'AuthController@destroy')->middleware('auth:api')->name('logout');
-Route::get('/user', 'AuthController@index')->middleware('auth:api')->name('user.current');
-
-// Admin
-Route::patch('/users/{user}/admin', 'AdminController@update')->middleware('auth:api', 'role:admin')->name('users.update');
-
-// hasUserSettings
-Route::post('/users/{user}/settings', 'SettingsController@store')->middleware('auth:api')->name('settings.personal');
-Route::post('/admins/{admin}/settings', 'SiteSettingsController@store')->middleware('auth:api', 'role:admin')->name('settings.global');
-
-Route::middleware(['auth:api'])->group(function () {
-    Route::post('/posts', 'PostsController@store')->name('post.create');
-    Route::delete('/posts/{post}', 'PostsController@destroy')->name('post.delete');
-    Route::get('/posts', 'PostsController@index')->name('post.getAll');
-    Route::get('/posts/{post}', 'PostsController@show')->name('post.getOne');
-    Route::put('/posts/{post}', 'PostsController@update')->name('post.update');
+Route::group(['prefix' => '/auth'], function () {
+    Route::post('/login', 'LoginController@login');
+    Route::post('/register', 'AuthController@store');
+    Route::post('/logout', 'AuthController@destroy')->middleware('auth:api');
+    Route::get('/user', 'AuthController@index')->middleware('auth:api');
+});
+
+
+// User Management system
+Route::group(['prefix' => '/users'], function () {
+    Route::patch('/{user}/admin', 'AdminController@update')->middleware('auth:api', 'role:admin');
+});
+
+// Settings management system
```

Comment:
This is not  the correct way to handle this

create a namespace called `user` and add these methods
`user/info`, get and post `user/settings`
and
`settings` namespace which should be only accessible by administrator

Reply:
I have made a /auth and now I will add settings to it to

Reply:
I meant it to be this way ,
We have a group for user management (sign up, sign in , info , sign out ) 
And then a route for admins to set site global settings
and a route for every user to store it's preference 
Why do you think it is wrong ?

Reply:
> We have a group for user management (sign up, sign in , info , sign out )
this is called the authentication group and the user settings should be inside aswell

admin is just a role, not less and not more. your routes should not be specified for the role. it should be declared for the feature

secondly we dont need the {user} here at all